Refresher/Patching/Patcher.cs

```diff
@@ -123,74 +139,151 @@ private static List<PatchTargetInfo> FindUrl(Stream file)
+            if (UrlMatch().Matches(str).Count != 0)
+                foundUrls.Add(new PatchTargetInfo
                 {
                     Length = len,
                     Offset = foundPosition,
                     Data = str,
                 });
         }
 
-        return found;
+        List<PatchTargetInfo> foundDigests = new();
+
+        foreach (long foundPosition in foundPossibleCookiePositions)
+        {
+            file.Position = foundPosition;
+
+            byte[] cookieBuf = new byte[8];
+            //If we didnt read enough or what we read isnt "cookie\0\0"
+            if (reader.Read(cookieBuf) < cookieBuf.Length || !"cookie\0\0"u8.SequenceEqual(cookieBuf))
+            {
+                //Skip this possibility
+                continue;
+            }
+
+            const int checkSize = 1000;
+
+            //Go back half the check size in bytes (so that `cookie` is in the middle)
+            file.Position -= checkSize / 2;
+
+            byte[] checkArr = new byte[checkSize];
+            Span<byte> toCheck = checkArr.AsSpan().Slice(0, reader.Read(checkArr));
+
+            Regex regex = DigestMatch();
+            for (int i = 0; i < toCheck.Length; i++)
+            {
+                //Skip all null bytes
+                while (i < toCheck.Length && toCheck[i] == 0) i++;
+
+                int len = 0;
+                int start = i;
+
+                //Go over all non-null bytes
+                while (i < toCheck.Length && toCheck[i] != 0)
+                {
+                    len++;
+                    i++;
+                }
+
+                if (len != 18)
+                    continue;
+
+                string str = Encoding.UTF8.GetString(toCheck.Slice(start, len));
+
+                //If theres exactly 18 matches,
+                if (regex.Matches(str).Count == 18)
+                {
+                    //Then we have found a digest key
+                    foundDigests.Add(new PatchTargetInfo
+                    {
+                        Length = len,
+                        Offset = file.Position - checkSize + start,
+                        Data = str,
+                    });
+                }
+            }
+        }
+
+        return (foundUrls, foundDigests);
     }
 
     /// <summary>
     ///     Checks the contents of the EBOOT to verify that it is patchable.
     /// </summary>
     /// <returns>A list of issues and notes about the EBOOT.</returns>
     [Pure]
-    public List<Message> Verify(string url)
+    public List<Message> Verify(string url, bool patchDigest)
     {
         // TODO: check if this is an ELF, correct architecture, if url is correct length, etc.
         List<Message> messages = new();
-        
+
+        this.Stream.Position = 0;
+        Class output = ELFReader.CheckELFType(this.Stream);
+        if (output == Class.NotELF)
+        {
+            messages.Add(new Message(MessageLevel.Error, "File is not a valid ELF!"));
+        }
```

Comment:
Also, this should be a warning. We shouldn't stop the patching process unless we absolutely have to.